admin/adminActionsRestricted.php

```diff
@@ -537,6 +542,53 @@ public function takeForumModerator(array $params)
 
 		return l_t('This user had their forum moderator status taken.');
 	}
+
+	public function changeUsername(array $params)
+	{
+		global $DB;
+		global $User;
+		
+		$userID = (int)$params['userID'];
+		$newUsername = (string)$params['username'];
+
+		if( !isset($params['reason']) || strlen($params['reason'])==0 )
+		{
+			return "Could not change username because no reason was given.";
+		}
+
+		$changeReason = $DB->msg_escape($params['reason']);
+
+		// check if username exists
+		list($result) = $DB->sql_row("SELECT username FROM wD_Users WHERE id = '".$userID."' AND username = '".$newUsername."'");
+		if (!empty($result)) 
+		{
+			return "This username has already been taken by another user.";
+		}
+		
+		// get and store old username, set new username
+		list($oldUsername) = $DB->sql_row("SELECT username FROM wD_Users WHERE id = ".$userID);

```

Comment:
Can you add a check here to make sure that an old username exists, it'll serve as a check to make sure a valid userID was given.